lib/marc_match_key/author_key.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+### A collection of functions and classes to generate a match key based on the
+###   GoldRush algorithm
+module MarcMatchKey
+  ### Generates the author portion of a GoldRush key
+  ### GoldRush includes the 130 field, even though that is not an author
+  ### Goldrush documentation says the key is padded to 5 characters, but it
+  ###   also says that it's padded to 20 characters
+  class AuthorKey
+    include MarcMatchFunctions
+    attr_reader :record, :key
+
+    def initialize(record)
+      @record = record
+      @key ||= generate_key
```

Comment:
To simplify, could we rename `generate_key` to `key` and make it public?  Then we wouldn't need to set the variable in the constructor or need to add the `attr_reader`.

If you still want to memoize it, you could do something like

```
  def key
    @key ||= begin
      auth_fields = record.fields.select do |field|
        %w[100 110 111 113 130].include?(field.tag) &&
          field['a']
      end
      author_key = ''.dup
      auth_fields.each do |field|
        part = clean_string(field['a'])
        author_key << part
      end
      pad_with_underscores(author_key, 20)
    end
  end
```

I think this would apply to most of the classes here.